### PR TITLE
Remove options.editorconfig from configuration.md

### LINKED ISSUE #15255: Correct documentation on the `editorconfig` option.
The issue is related to #6176

The documentation describes a `editorconfig` option, and trying to implement results in an error (tried both ways):
```
[warn] Ignored unknown option { editorconfig: true }.
[warn] Ignored unknown option { options: { editorconfig: true } }.
```
#6176 Says there is no `editorconfig` option, please update the documentation: 
<https://prettier.io/docs/en/configuration#editorconfig>
> If **options.editorconfig** is true and an [.editorconfig file](https://editorconfig.org/) is in your project, Prettier will parse it and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by .prettierrc, etc.

### PR DESCRIPTION
**Pull request by cavcrosby**
## Description

Fixes #15255.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -200,7 +200,7 @@ If you’d like a JSON schema to validate your configuration, one is available h
 
 ## EditorConfig
 
-If `options.editorconfig` is `true` and an [`.editorconfig` file](https://editorconfig.org/) is in your project, Prettier will parse it and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by `.prettierrc`, etc.
+If an [`.editorconfig` file](https://editorconfig.org/) is in your project, Prettier will parse it and convert its properties to the corresponding Prettier configuration. This configuration will be overridden by `.prettierrc`, etc.
```
Comment (sosukesuzuki) [inline docs/configuration.md:203]: I would also like you to mention that you can prevent this behavior by using https://prettier.io/docs/en/cli#--no-editorconfig.

**Comment by cavcrosby** [comment]
Is there anything further adjustments you wish me to make @sosukesuzuki?

**Review by fisker — APPROVED** [review]

**Comment by cavcrosby** [comment]
Just rebased my changes, giving things another nudge to see this through.

**Comment by cavcrosby** [comment]
This was fixed in #16335.

**Pull request closed without merging**